lib/_http_incoming.js

```diff
@@ -114,19 +114,21 @@ IncomingMessage.prototype._read = function _read(n) {
     readStart(this.socket);
 };
 
-
 // It's possible that the socket will be destroyed, and removed from
 // any messages, before ever calling this.  In that case, just skip
 // it, since something else is destroying this connection anyway.
-IncomingMessage.prototype.destroy = function destroy(error) {
-  // TODO(ronag): Implement in terms of _destroy
-  this.destroyed = true;
-  if (this.socket)
-    this.socket.destroy(error);
-  return this;
+IncomingMessage.prototype._destroy = function _destroy(err, cb) {
+  // TODO (dnlup): move abort logic here based on readableEnded state
+  if (!this.readableEnded || !this.complete) {
```

Comment:
I tried to move the abort logic here. To make it work for both client and server I also had to check for the `readableEnded` state.